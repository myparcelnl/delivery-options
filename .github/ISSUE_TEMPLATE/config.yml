blank_issues_enabled: false
contact_links:
  - name: Try the sandbox
    url: https://myparcelnl.github.io/delivery-options/
    about: The sandbox contains all available options and allows you to build or test configurations.
  - name: Contact customer support
    url: https://www.myparcel.nl/services/contact/
    about: For issues regarding your account and shipments.
