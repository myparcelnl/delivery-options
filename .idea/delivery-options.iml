<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager">
    <content url="file://$MODULE_DIR$">
      <excludeFolder url="file://$MODULE_DIR$/dist" />
      <excludeFolder url="file://$MODULE_DIR$/.yarn" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
      <excludeFolder url="file://$MODULE_DIR$/artifact" />
      <excludeFolder url="file://$MODULE_DIR$/.husky" />
      <excludeFolder url="file://$MODULE_DIR$/.nx" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>