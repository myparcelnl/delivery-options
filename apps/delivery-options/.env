VITE_CLASS_BASE=myparcel-delivery-options
