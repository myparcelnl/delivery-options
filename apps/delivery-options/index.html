<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta
      content="width=device-width, initial-scale=1.0"
      name="viewport" />
    <title>MyParcel Delivery Options</title>
  </head>
  <body>
    <script type="importmap">
      {
        "imports": {
          "vue": "https://unpkg.com/vue@3/dist/vue.esm-browser.js"
        }
      }
    </script>

    <div id="myparcel-delivery-options"></div>

    <script src="/src/main.ts" type="module"></script>
  </body>
</html>
