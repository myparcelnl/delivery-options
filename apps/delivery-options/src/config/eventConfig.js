/**
 * ConfigBus events.
 *
 * These are only used for and by the configBus so they don't need to be namespaced.
 */

/**
 * Sent by configBus when a setting is updated.
 *
 * @type {String}
 */
export const UPDATE = 'update';

/**
 * Sent by configBus after a setting is updated (by `UPDATE`).
 *
 * @type {String}
 */
export const AFTER_UPDATE = 'after_update';

/**
 * Sent by configBus when an error has occurred.
 *
 * @type {String}
 */
export const ERROR = 'error';

/*
 * Document events
 * The following events should be namespaced to avoid any possible conflicts.
 */

/**
 * This event is used to change the config after the initialization.
 *
 * @type {String}
 */
export const UPDATE_CONFIG_IN = 'myparcel_update_config';

/**
 * For the external platform to tell this application to update. If it's a CustomEvent the detail will be used as the
 *  address.
 *
 * @type {String}
 */
export const UPDATE_DELIVERY_OPTIONS = 'myparcel_update_delivery_options';

/**
 * To tell the external platform the address is updated.
 *
 * @type {String}
 */
export const UPDATED_ADDRESS = 'myparcel_updated_address';

/**
 * To tell the external platform it needs to update.
 *
 * @type {String}
 */
export const UPDATED_DELIVERY_OPTIONS = 'myparcel_updated_delivery_options';

/**
 * Disable the delivery options.
 *
 * @type {String}
 */
export const DISABLE_DELIVERY_OPTIONS = 'myparcel_disable_delivery_options';

/**
 * Manually show the delivery options. The update listener has to be re-enabled after this.
 *
 * @type {String}
 */
export const SHOW_DELIVERY_OPTIONS = 'myparcel_show_delivery_options';

/**
 * Manually hide the delivery options. Should remove the update listener.
 *
 * @type {String}
 */
export const HIDE_DELIVERY_OPTIONS = 'myparcel_hide_delivery_options';

/**
 * Used to (re)render the entire module.
 *
 * @type {String}
 */
export const RENDER_DELIVERY_OPTIONS = 'myparcel_render_delivery_options';
