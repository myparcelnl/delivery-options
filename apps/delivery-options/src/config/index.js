export * from './appConfig';
export * from './configBus';
export * from './configurationMerger';
export * from './defaultConfiguration';
export * from './defaultStrings';
export * from './errorConfig';
export * from './eventConfig';
export * from './exports';
export * from './extraDeliveryConfig';
export * from './formConfig';
export * from './getAddress';
export * from './getConfig';
export * from './getWindowObject';
export * from './hasFilterableValue';
export * from './isPastTime';
export * from './localeConfig';
export * from './platform';
export * from './settingHasCarrierOverride';
export * from './settingHasCountryOverride';
export * from './urlConfig';
export * from './validatePlatform';
