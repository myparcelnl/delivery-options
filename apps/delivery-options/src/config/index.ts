export * from './filterConfig';
export * from './getDefaultConfigForPlatform';
export * from './getDefaultStrings';
export * from './handleDeprecatedOptions';
export * from './setConfiguration';
export * from './validateConfig';
