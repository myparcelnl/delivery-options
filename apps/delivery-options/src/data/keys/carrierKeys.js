/** @type {MyParcel.CarrierName} */
export const POSTNL = 'postnl';

/** @type {MyParcel.CarrierName} */
export const DPD = 'dpd';

/** @type {MyParcel.CarrierName} */
export const BPOST = 'bpost';

/** @type {MyParcel.CarrierName} */
export const INSTABOX = 'instabox';

/** @type {MyParcel.CarrierName} */
export const CHEAP_CARGO = 'cheapcargo';

/** @type {MyParcel.CarrierName} */
export const DHL = 'dhl';

/** @type {MyParcel.CarrierName} */
export const DHL_FOR_YOU = 'dhlforyou';

/** @type {MyParcel.CarrierName} */
export const DHL_PARCEL_CONNECT = 'dhlparcelconnect';

/** @type {MyParcel.CarrierName} */
export const DHL_EUROPLUS = 'dhleuroplus';
