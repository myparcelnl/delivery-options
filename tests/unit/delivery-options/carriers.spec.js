import * as CARRIERS from '@/data/keys/carrierKeys';
import * as CONFIG from '@/data/keys/configKeys';
import * as PLATFORMS from '@/data/keys/platformKeys';
import { UPDATED_DELIVERY_OPTIONS } from '@/config/eventConfig';
import { merge } from 'lodash-es';
import { mockDeliveryOptions } from '@Tests/unit/delivery-options/mockDeliveryOptions';
import { waitForEvent } from '@Tests/waitForEvent';

const CONFIG_SINGLE_CARRIER = Object.freeze({
  [CONFIG.KEY]: {
    [CONFIG.PLATFORM]: PLATFORMS.MYPARCEL,
    [CONFIG.CARRIER_SETTINGS]: {
      [CARRIERS.POSTNL]: {
        [CONFIG.ALLOW_DELIVERY_OPTIONS]: true,
      },
    },
  },
});

const CONFIG_MULTIPLE_CARRIERS = Object.freeze(merge({}, CONFIG_SINGLE_CARRIER, {
  [CONFIG.KEY]: {
    [CONFIG.CARRIER_SETTINGS]: {
      [CARRIERS.DHL_FOR_YOU]: {
        [CONFIG.ALLOW_DELIVERY_OPTIONS]: true,
      },
    },
  },
}));

describe('carriers', () => {
  it('shows one carrier', async() => {
    expect.assertions(3);
    const app = await mockDeliveryOptions(CONFIG_SINGLE_CARRIER);

    await waitForEvent(UPDATED_DELIVERY_OPTIONS);
    const postnl = app.findChoice('carrier', CARRIERS.POSTNL);
    expect(app.findChoice('carrier', CARRIERS.DHL_FOR_YOU).exists()).toBe(false);
    expect(postnl.exists()).toBe(true);
    expect(app.findChoice('carrier', CARRIERS.DHL).exists()).toBe(false);
  });

  it('can show multiple carriers', async() => {
    expect.assertions(5);
    const app = await mockDeliveryOptions(CONFIG_MULTIPLE_CARRIERS);

    await waitForEvent(UPDATED_DELIVERY_OPTIONS);

    const postNl = app.findChoice('carrier', CARRIERS.POSTNL);
    const dhlForYou = app.findChoice('carrier', CARRIERS.DHL_FOR_YOU);
    const bpost = app.findChoice('carrier', CARRIERS.BPOST);

    expect(postNl.exists()).toBe(true);
    expect(dhlForYou.exists()).toBe(true);
    expect(bpost.exists()).toBe(false);

    expect(postNl.find('[data-test-id="image"]').exists()).toBe(true);
    expect(dhlForYou.find('[data-test-id="image"]').exists()).toBe(true);
  });

  it('does not show a carrier logo when there is only one carrier', async() => {
    const app = await mockDeliveryOptions(CONFIG_SINGLE_CARRIER);

    await waitForEvent(UPDATED_DELIVERY_OPTIONS);

    const postNl = app.findChoice('carrier', CARRIERS.POSTNL);

    expect(postNl.find('[data-test-id="image"]').exists()).toBe(false);
  });

  it('shows carrier logos when there are multiple carriers', async() => {
    expect.assertions(2);
    const app = await mockDeliveryOptions(CONFIG_MULTIPLE_CARRIERS);

    await waitForEvent(UPDATED_DELIVERY_OPTIONS);

    const postNl = app.findChoice('carrier', CARRIERS.POSTNL);
    const dhlForYou = app.findChoice('carrier', CARRIERS.DHL_FOR_YOU);

    expect(postNl.find('[data-test-id="image"]').exists()).toBe(true);
    expect(dhlForYou.find('[data-test-id="image"]').exists()).toBe(true);
  });
});
