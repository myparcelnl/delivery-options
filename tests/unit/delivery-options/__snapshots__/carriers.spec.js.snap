// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`carriers does not show a carrier logo when there is only one carrier 1`] = `
<form class="myparcel-delivery-options">
  <!---->
  <div>
    <!---->
    <table class="myparcel-delivery-options__table" data-test-id="delivery">
      <tr class="myparcel-delivery-options__choice myparcel-delivery-options__choice--deliver myparcel-delivery-options__choice--selected" data-test-id="delivery" data-test-choice="deliver">
        <td class="myparcel-delivery-options__input">
          <div><input id="myparcel-delivery-options__delivery--deliver" type="radio" value="deliver" data-test-id="delivery__input" data-test-choice="deliver"></div>
        </td>
        <td colspan="2"><label for="myparcel-delivery-options__delivery--deliver"><span>Thuis of op het werk bezorgen</span>
            <!---->
            <!----></label>
          <transition-group-stub tag="div" name="fade" appear="true">
            <table class="myparcel-delivery-options__table">
              <tr class="myparcel-delivery-options__choice myparcel-delivery-options__choice--postnl myparcel-delivery-options__choice--selected" data-test-id="carrier" data-test-choice="postnl">
                <!---->
                <td><label for="myparcel-delivery-options__carrier--postnl"><span></span>
                    <!---->
                    <!----></label>
                  <transition-group-stub tag="div" name="fade" appear="true">
                    <table class="myparcel-delivery-options__table">
                      <tr>
                        <td><label><select id="myparcel-delivery-options__deliveryDate" name="deliveryDate" class="myparcel-delivery-options__w-100" data-test-id="deliveryDate__select">
                              <option value="2023-03-07T00:00:00.000Z" data-test-id="deliveryDate__select__option" data-test-choice="2023-03-07T00:00:00.000Z">Dinsdag 7 maart</option>
                              <option value="2023-03-08T00:00:00.000Z" data-test-id="deliveryDate__select__option" data-test-choice="2023-03-08T00:00:00.000Z">Woensdag 8 maart</option>
                              <option value="2023-03-09T00:00:00.000Z" data-test-id="deliveryDate__select__option" data-test-choice="2023-03-09T00:00:00.000Z">Donderdag 9 maart</option>
                              <option value="2023-03-10T00:00:00.000Z" data-test-id="deliveryDate__select__option" data-test-choice="2023-03-10T00:00:00.000Z">Vrijdag 10 maart</option>
                              <option value="2023-03-11T00:00:00.000Z" data-test-id="deliveryDate__select__option" data-test-choice="2023-03-11T00:00:00.000Z">Zaterdag 11 maart</option>
                              <option value="2023-03-14T00:00:00.000Z" data-test-id="deliveryDate__select__option" data-test-choice="2023-03-14T00:00:00.000Z">Dinsdag 14 maart</option>
                              <option value="2023-03-15T00:00:00.000Z" data-test-id="deliveryDate__select__option" data-test-choice="2023-03-15T00:00:00.000Z">Woensdag 15 maart</option>
                            </select></label></td>
                      </tr>
                    </table>
                    <table class="myparcel-delivery-options__table">
                      <tr class="myparcel-delivery-options__choice myparcel-delivery-options__choice--morning" data-test-id="deliveryMoment" data-test-choice="morning">
                        <td class="myparcel-delivery-options__input">
                          <div><input id="myparcel-delivery-options__deliveryMoment--morning" type="radio" value="morning" data-test-id="deliveryMoment__input" data-test-choice="morning"></div>
                        </td>
                        <td colspan="1"><label for="myparcel-delivery-options__deliveryMoment--morning"><span>08:00 – 12:00</span>
                            <!----> <span class="myparcel-delivery-options__float--right" data-test-id="price"><!----></span></label>
                          <!---->
                        </td>
                      </tr>
                      <tr class="myparcel-delivery-options__choice myparcel-delivery-options__choice--standard myparcel-delivery-options__choice--selected" data-test-id="deliveryMoment" data-test-choice="standard">
                        <td class="myparcel-delivery-options__input">
                          <div><input id="myparcel-delivery-options__deliveryMoment--standard" type="radio" value="standard" data-test-id="deliveryMoment__input" data-test-choice="standard"></div>
                        </td>
                        <td colspan="1"><label for="myparcel-delivery-options__deliveryMoment--standard"><span>08:30 – 21:30</span>
                            <!----> <strong class="myparcel-delivery-options__float--right" data-test-id="price"><!----></strong></label>
                          <!---->
                        </td>
                      </tr>
                      <tr class="myparcel-delivery-options__choice myparcel-delivery-options__choice--evening" data-test-id="deliveryMoment" data-test-choice="evening">
                        <td class="myparcel-delivery-options__input">
                          <div><input id="myparcel-delivery-options__deliveryMoment--evening" type="radio" value="evening" data-test-id="deliveryMoment__input" data-test-choice="evening"></div>
                        </td>
                        <td colspan="1"><label for="myparcel-delivery-options__deliveryMoment--evening"><span>18:00 – 22:00</span>
                            <!----> <span class="myparcel-delivery-options__float--right" data-test-id="price"><!----></span></label>
                          <!---->
                        </td>
                      </tr>
                      <!---->
                    </table>
                    <table class="myparcel-delivery-options__table">
                      <tr class="myparcel-delivery-options__choice myparcel-delivery-options__choice--only_recipient" data-test-id="shipmentOptions" data-test-choice="only_recipient">
                        <td class="myparcel-delivery-options__input">
                          <div><input id="myparcel-delivery-options__shipmentOptions--only_recipient" type="checkbox" name="shipmentOptions" value="only_recipient" data-test-id="shipmentOptions__input" data-test-choice="only_recipient"></div>
                        </td>
                        <td colspan="1"><label for="myparcel-delivery-options__shipmentOptions--only_recipient"><span>Alleen ontvanger</span>
                            <!----> <span class="myparcel-delivery-options__float--right" data-test-id="price"><!----></span></label>
                          <!---->
                        </td>
                      </tr>
                      <tr class="myparcel-delivery-options__choice myparcel-delivery-options__choice--signature" data-test-id="shipmentOptions" data-test-choice="signature">
                        <td class="myparcel-delivery-options__input">
                          <div><input id="myparcel-delivery-options__shipmentOptions--signature" type="checkbox" name="shipmentOptions" value="signature" data-test-id="shipmentOptions__input" data-test-choice="signature"></div>
                        </td>
                        <td colspan="1"><label for="myparcel-delivery-options__shipmentOptions--signature"><span>Handtekening voor ontvangst</span>
                            <!----> <span class="myparcel-delivery-options__float--right" data-test-id="price"><!----></span></label>
                          <!---->
                        </td>
                      </tr>
                      <!---->
                    </table>
                  </transition-group-stub>
                </td>
              </tr>
              <!---->
            </table>
          </transition-group-stub>
        </td>
      </tr>
      <tr class="myparcel-delivery-options__choice myparcel-delivery-options__choice--pickup" data-test-id="delivery" data-test-choice="pickup">
        <td class="myparcel-delivery-options__input">
          <div><input id="myparcel-delivery-options__delivery--pickup" type="radio" value="pickup" data-test-id="delivery__input" data-test-choice="pickup"></div>
        </td>
        <td colspan="2"><label for="myparcel-delivery-options__delivery--pickup"><span>Afhalen op locatie</span> <span class="myparcel-delivery-options__float--right" data-test-id="priceTag">Vanaf €&nbsp;0,00</span>
            <!----></label>
          <!---->
        </td>
      </tr>
      <!---->
    </table>
  </div>
</form>
`;
